src/js/components/add-journal-entry/journal-entry.js

```diff
@@ -69,6 +100,12 @@ class JournalEntry extends MaharaBaseComponent {
         //console.log("new tags", tags);
         this.tags = tags; // parent component accesses it this way
     }
+
+    changeJournal() {
+        let targetBlogId = parseInt(this.refs.journalSelect.el.select2('data')[0].id, 10);
```

Comment:
that's a way spec in library https://github.com/rkit/react-select2-wrapper/tree/master/guide 'Access to select2' section